### DISCUSSION TITLE
What are the all list of plugins supported with Filament Rich Text editor v3?

### DISCUSSION THREAD

**Discussion by subhadipghorui** [filamentphp/filament#11345] [Help]
### Package

Panel builder

### Package Version

v.3.2.0

### How can we help you?

use Filament\Forms\Components\RichEditor;
 
RichEditor::make('content')
    ->toolbarButtons([
        'attachFiles',
        'blockquote',
        'bold',
        'bulletList',
        'codeBlock',
        'h2',
        'h3',
        'italic',
        'link',
        'orderedList',
        'redo',
        'strike',
        'underline',
        'undo',
    ])

**1. What are the all list of plugins supported with Filament Rich Text editor v3?
2. What about license ? can we use this for commercial app making for clients**

**Comment by kpsanmao** [comment]
So did I.

**Comment by W1-PopelierE** [comment]
Hi, 

1.
Listing all toolbarButtons can be done by logging `Forms\Components\RichEditor::make(')->getToolbarButtons()`.
Or take a look at this file https://github.com/filamentphp/filament/blob/3.x/packages/forms/src/Components/RichEditor.php

2.
FilamentPHP uses Trix-editor. Trix has a MIT license which allows you to use it in any setting (commercial or not). 
https://github.com/basecamp/trix?tab=MIT-1-ov-file#readme

**Comment by kpsanmao** [comment]
Thanks your reply。
The Trix-editor don't support some common operations，like text alignment。It's more like an immature product。
I used https://filamentphp.com/plugins/amid-tinyeditor and i think official rich-editor  should support more。
[https://filamentphp.com/images/content/plugins/images/amid-tinyeditor.webp]<https://filamentphp.com/plugins/amid-tinyeditor>
TinyEditor by Amid Esfahani - Filament<https://filamentphp.com/plugins/amid-tinyeditor>
A Tinymce 6 Rich Text Editor plugin for Filament Forms.
filamentphp.com

________________________________
发件人: Emiel Popelier ***@***.***>
发送时间: 2024年3月20日 16:32
收件人: filamentphp/filament ***@***.***>
抄送: kpsanmao ***@***.***>; Comment ***@***.***>
主题: Re: [filamentphp/filament] What are the all list of plugins supported with Filament Rich Text editor v3? (Discussion #11345)


Hi,

  1.

Listing all toolbarButtons can be done by logging Forms\Components\RichEditor::make(')->getToolbarButtons().
Or take a look at this file https://github.com/filamentphp/filament/blob/3.x/packages/forms/src/Components/RichEditor.php

  1.

FilamentPHP uses Trix-editor. Trix has a MIT license which allows you to use it in any setting (commercial or not).
https://github.com/basecamp/trix?tab=MIT-1-ov-file#readme

―
